packages/core/utils/src/common/normalize-import-path-with-source.ts

```diff
@@ -16,9 +16,15 @@ export function normalizeImportPathWithSource(
    * happen under the hood.
    */
   if (normalizePath?.startsWith("./")) {
-    const sourceDir = process[Symbol.for("ts-node.register.instance")]
-      ? "src"
-      : "dist"
+    /**
+     * If someone is using the correct path pointing to the "src" directory
+     * then we are all good. Otherwise we will point to the "src" directory.
+     *
+     * In case of the production output. The app should be executed from within
+     * the "./build" directory and the "./build" directory will have the
+     * "./src" directory inside it.
+     */
+    let sourceDir = normalizePath.startsWith("./src") ? "./" : "./src"
```

Comment:
q: shouldn't this always be `src`?

Reply:
In the existing projects, the module resolve string looks as `modules/hello`, basically not including the `src` prefix. In this check, I am making sure that we do not break those existing projects in development. 

Lemme know if its still not clear, we can do a huddle

Reply:
Ah yes ofc, that's nice.